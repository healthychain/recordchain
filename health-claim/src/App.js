import React from "react";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import Login from "./components/Login/Login";
import Claim from "./components/Claim/Claim";
import NotFound from "./components/NotFound/NotFound";
import "./App.css";
import DoctorContainer from "./containers/DoctorContainer";

export class App extends React.PureComponent {
  render() {
    return (
      <div className="App container">
        <Router>
          <Switch>
            <Route path="/" exact component={Login} />
            <Route path="/patient" exact component={Claim} />
            <Route path="/doctor" exact component={DoctorContainer} />
            <Route path="/login" exact component={Login} />
            {/* Finally, catch all unmatched routes */}
            <Route component={NotFound} />\
          </Switch>
        </Router>
      </div>
    );
  }
}

export default App;
