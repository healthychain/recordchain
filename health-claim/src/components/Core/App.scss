// App.scss

// Scaffo
html {
  background: #f2f2f2;
}
.Body {
  background: #f2f2f2;
}

.Main {
  max-width: 960px;
  margin: 0 auto;
  padding-bottom: 96px;
}
.Page__Title {
  padding-top: 1.67em;
  margin-bottom: 0.33em;
  position: relative;
  display: block;
  .Page__Title--deadline {
    float: right;
    position: absolute;
    font-size: 0.67em;
    padding-top: 0.51em;
    line-height: 1;
    right: 0;
    opacity: 0.7;
  }
}
.Page__Subtitle {
  margin: 0;
}
.Section__Header {
  font-size: 16px;
  text-transform: uppercase;
  font-weight: bold;
  color: rgba(48, 94, 212, 1);
  padding-top: 1.7em;
  padding-bottom: 0.5em;
  margin-bottom: 0.67em;
  border-bottom: 1px solid #e0e0e0;
}

// Temp
.Header {
  display: block;
  left: 0;
  position: relative;
  top: 0;
}
.Header__Layout {
  display: flex;
  width: 100%;
  flex-direction: row;
  justify-content: space-between;
}

.Header__Upper {
  background: rgb(28, 29, 92);
  height: 56px;
  top: 0;
  z-index: 99;
  position: fixed;
  width: 100%;
  .Header__Nav {
    align-items: center;
    display: flex;
    height: 56px;
    margin-left: 48px;
    margin-right: 48px;
    max-width: 960px;
  }
  .Header__Link,
  .Header__Logo {
    color: #fff;
    font-weight: 600;
    margin: 0 1vw;
    & > a {
      color: #fff;
      text-decoration: none;
    }
  }
  .Header__Logo {
    opacity: 0.66;
  }
}
.Header__Lower {
  align-items: center;
  background: rgba(48, 94, 212, 1);
  display: flex;
  height: 56px;
  padding-left: 48px;
  padding-right: 48px;
  margin-top: 56px;
  .Header__Title {
    color: #fff;
    line-height: 1;
    margin-top: -4px;
    margin-bottom: 0;
    margin-left: 1vw;
    font-size: 1em;
  }
}
