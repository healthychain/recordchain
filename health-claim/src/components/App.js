import React from "react";
import { BrowserRouter as Router, Route, Switch, Link } from "react-router-dom";
import Claim from "../Claim/Claim";
import NotFound from "../NotFound/NotFound";
import Home from "../Home";
import "./App.scss";
import DoctorContainer from "../../containers/DoctorContainer";
import LoginFormContainer from "../../containers/LoginFormContainer";
import RegisterFormContainer from "../../containers/RegisterFormContainer";
import MessageScreen from "../MessageScreen/MessageScreen";

export class App extends React.PureComponent {
  render() {
    return (
      <Router>
        <>
          <div className="Header">
            <div className="Header__Upper">
              <div className="Header__Nav">
                <div className="Header__Logo">HealthClaim</div>{" "}
                <div className="Header__Link">
                  <Link to="/" onClick={() => this.props.logout()}>
                    Logout
                  </Link>
                </div>
              </div>
            </div>
            <div className="Header__Lower">
              <h2 className="Header__Title">Decentralized Healthcare</h2>
            </div>
          </div>
          <div className="Body">
            <div className="Main">
              <Switch>
                <Route path="/" exact component={Home} />
                <Route
                  path="/login-doctor"
                  exact
                  render={() => <LoginFormContainer userType="Doctor" />}
                />
                <Route
                  path="/login-patient"
                  exact
                  render={() => <LoginFormContainer userType="Patient" />}
                />
                <Route
                  path="/register"
                  exact
                  render={() => <RegisterFormContainer />}
                />
                <Route
                  path="/register-success"
                  exact
                  render={() => (
                    <MessageScreen
                      success
                      redirect="/login-doctor"
                      title="You're all set!"
                      text="You have successfully registered."
                      buttonText="Log in"
                    />
                  )}
                />
                <Route
                  path="/register-failure"
                  exact
                  render={() => (
                    <MessageScreen
                      redirect="/"
                      title="Something went wrong"
                      text="Please try again"
                      buttonText="Home"
                    />
                  )}
                />
                <Route path="/patient" exact component={Claim} />
                <Route path="/doctor" exact component={DoctorContainer} />
                {/* Finally, catch all unmatched routes */}
                <Route component={NotFound} />\
              </Switch>
            </div>
          </div>
        </>
      </Router>
    );
  }
}

export default App;
