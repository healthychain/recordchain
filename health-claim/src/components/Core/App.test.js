import React from "react";
import renderer from "react-test-renderer";
import App from "./App";
describe("App", () => {
  it("should render without crashing", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
