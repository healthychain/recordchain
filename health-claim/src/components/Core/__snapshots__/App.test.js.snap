// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render without crashing 1`] = `
Array [
  <div
    className="Header"
  >
    <div
      className="Header__Upper"
    >
      <div
        className="Header__Nav"
      >
        <div
          className="Header__Layout"
        >
          <div
            className="Header__Logo"
          >
            HealthClaim
          </div>
           
        </div>
      </div>
    </div>
    <div
      className="Header__Lower"
    >
      <h2
        className="Header__Title"
      >
        Decentralized Healthcare
      </h2>
    </div>
  </div>,
  <div
    className="Body"
  >
    <div
      className="Main"
    >
      <h3
        className="Page__Title"
      >
        Welcome
      </h3>
      <div
        className="Box Box__NP"
      >
        <div
          className="Flex__Double"
        >
          <div
            className="Flex__Half Flex__Grey Flex__Separator"
          >
            <h2
              className="Page__Halftitle"
            >
              Log in
            </h2>
            <p
              className="Page__Text"
            >
              If you are a doctor or have already created an account, please log in below.
            </p>
            <p>
              <button
                className="Button Button__Green"
                onClick={[Function]}
              >
                Doctor
              </button>
                   
              <button
                className="Button Button__Green"
                onClick={[Function]}
              >
                Patient
              </button>
            </p>
          </div>
          <div
            className="Flex__Half"
          >
            <h2
              className="Page__Halftitle"
            >
              Sign up
            </h2>
            <p
              className="Page__Text"
            >
              Create your account now to keep your medical records safe.
            </p>
            <p
              className="Page__Text"
            >
              <button
                className="Button Button__Secondary"
                onClick={[Function]}
              >
                Sign up
              </button>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
